executor/set.go

```diff
@@ -204,16 +204,30 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
 	return nil
 }
 
-func (e *SetExecutor) checkPDClientDynamicOption(name, valStr string, sessionVars *variable.SessionVars) error {
-	var err error
+func (e *SetExecutor) checkPDClientDynamicOption(name string, sessionVars *variable.SessionVars) error {
+	if name != variable.TiDBTSOClientBatchMaxWaitTime &&
+		name != variable.TiDBEnableTSOFollowerProxy {
+		return nil
+	}
+	var (
+		err    error
+		valStr string
+	)
+	valStr, err = sessionVars.GlobalVarsAccessor.GetGlobalSysVar(name)
```

Comment:
Why cannot use the old `valStr`?

Reply:
Because the valve will be truncated if it's out of range, using the old value will fail when set it to the PD client.